index.js

```diff
@@ -316,4 +344,48 @@ function getBalance(node) {
 	return 0;
 }
 
-//TODO: union, intersection and set difference, helpers, Split and Join
\ No newline at end of file
+// join returns a copy instead of modifying the current trees
+AvlTree.prototype.join = function (element, node2) {
+	var tree1 = this.copy();
+	var tree2 = node2.copy();
+	
+	var keyNode = new Node(element);
+	
+	var heightTree1 = tree1._root.height;
+	var heightTree2 = tree2._root.height;
+	
+	if (Math.abs(heightTree1 - heightTree2) <= 1) {
+		var joinedTree = new AvlTree(this._compare);
+		keyNode.left = tree1._root;
+		keyNode.right = tree2._root;
+		updateHeight(keyNode);
+		joinedTree._root = keyNode;
+		return joinedTree;
+	}
+	
+	var joinedTree = new AvlTree(this._compare);
+	joinedTree._root = tree1._root;
+	var currNode;
+	var parentNode;
+	
+	if (heightTree1 > heightTree2) {
+		currNode = tree1._root;
+		
+		while (currNode.height > heightTree2 + 1) {
+			parentNode = currNode;
+			currNode = currNode.right;
+		}
+		
+		keyNode.left = currNode;
+		keyNode.right = tree2._root;
+		updateHeight(keyNode);
+		parentNode.right = keyNode;
+		
+		joinedTree._balance(joinedTree._root, null);
+		updateHeight(joinedTree._root);
+	}
+	
```

Comment:
What about heightTree1 > heightTree2 ?

Reply:
I... overlooked this. Sorry about that! The Wikipedia algorithm was admittedly a little confusing and I had to check various sources for how they'd go about it, and somewhere along the way, this slipped through the cracks. This definitely needs to be implemented too.